src/azure-cli/azure/cli/command_modules/rdbms/validators.py

```diff
@@ -313,14 +313,15 @@ def pg_arguments_validator(db_context, location, tier, sku_name, storage_gb, ser
             resource_group=parse_resource_id(instance.id)["resource_group"],
             server_name=instance.name)
     sku_info = list_location_capability_info['sku_info']
+    sku_info = {k.lower(): v for k, v in sku_info.items()}
     single_az = list_location_capability_info['single_az']
     geo_backup_supported = list_location_capability_info['geo_backup_supported']
     _network_arg_validator(subnet, public_access)
     _pg_tier_validator(tier, sku_info)  # need to be validated first
     if tier is None and instance is not None:
-        tier = instance.sku.tier
-    if "supported_storageV2_size" in sku_info[tier]:
-        supported_storageV2_size = sku_info[tier]["supported_storageV2_size"]
+        tier = instance.sku.tier.lower()
+    if "supported_storageV2_size" in sku_info[tier.lower()]:
+        supported_storageV2_size = sku_info[tier.lower()]["supported_storageV2_size"]
```

Comment:
tier is already set to instance.sku.tier.lower(), do you need to set lower again

Reply:
There's a possibility that tier is not in lower case when it is not in the if loop. I'd like to keep this lower case here.

Reply:
Oh I see, tier is an input for the method, and it is also being reset.